### PR TITLE
Refactor: MessageEventManager

### PR DESCRIPTION
**Pull request by uni-kakurenbo**
## Refactor
- Renamed `MessageEvent` to `MessageEventManager`.
- Made super class `BaseEventManager`.

## Lint
- Added some words to workspace settings: `scratchapi`, `scratchapijs`.

## Samples
- Moved the message event listener out of `ready` event listener.

## Other
- Remove a duplicate field, `name` in `package.json`.
- Made `MessageEventManager` clear its monitoring when the `client` is destroyed.

### CODE REVIEW

File: src/session/Session.js [hunk 1]
```diff
@@ -32,28 +39,29 @@ class Session {
...
+      { responseType: 'document' },
+    );
     if (response.isAxiosError) throw new Error('COMMENT_REJECTED');
     return response;
+
+    function parseOptions() {
+      let type, identifier;
+      if (options.project) {
+        type = 'project';
+        identifier = options.project;
```
Comment (itta611) [inline src/session/Session.js:55]: this can be outside of if statement.

**Review by itta611 — CHANGES_REQUESTED** [review]
Just a fix, please.

File: src/session/Session.js (hunk 1 above)
Comment (uni-kakurenbo) [reply src/session/Session.js:55]: What does the "this" mean?

Comment (uni-kakurenbo) [reply src/session/Session.js:55]: Oh, I got it.

Comment (uni-kakurenbo) [reply src/session/Session.js]: Is this what you meant?

Comment (itta611) [reply src/session/Session.js]: That is good!

nice job.

**Review by itta611 — APPROVED** [review]

**Pull request merged**